homeassistant/components/device_tracker/tplink.py

```diff
@@ -38,9 +38,13 @@ def get_scanner(hass, config):
     for cls in [Tplink5DeviceScanner, Tplink4DeviceScanner,
                 Tplink3DeviceScanner, Tplink2DeviceScanner,
                 TplinkDeviceScanner]:
-        scanner = cls(config[DOMAIN])
-        if scanner.success_init:
-            return scanner
+        try:
+            _LOGGER.info("Trying with " + cls.__name__)
+            scanner = cls(config[DOMAIN])
+            if scanner.success_init:
+                return scanner
+        except:
```

Comment:
It seems that most of the scanners are careful to catch exceptions and return `False`.

So maybe you can just handle the error that you have experienced and avoid this catch-all that will hide all kinds of real errors?